botocore/utils.py

```diff
@@ -35,6 +35,11 @@
 # These are chars that do not need to be urlencoded.
 # Based on rfc2986, section 2.3
 SAFE_CHARS = '-._~'
+LABEL_RE = re.compile('[a-z0-9][a-z0-9\-]*[a-z0-9]')
```

Comment:
But isn't the `*` for the `[a-z0-9\-]` block?  That would mean something like aa-a wouldn't match anymore.